### PR TITLE
Fix #1873 - Move Skip button below Continue button on Onboarding

### LINKED ISSUE #1873: Fix onboarding issue for Russian

### PR DESCRIPTION
**Pull request by Brandon-T**
## Summary of Changes

This pull request fixes issue #1873 
- Move Skip button below Continue button on Onboarding (Vertical Alignment)

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by Brandon-T** [comment]
I added some screenshots.. The one where I tinted the background colour is just to show how much spacing is there and also the area of the skip button so that we can see properly the layout.

Screen Shot 2019-11-02 at 9 10 07 AM
Screen Shot 2019-11-02 at 9 03 32 AM
Screen Shot 2019-11-02 at 9 03 24 AM

**Comment by Brandon-T** [comment]
Screen Shot 2019-11-02 at 9 13 33 AM
Screen Shot 2019-11-02 at 9 13 26 AM
Screen Shot 2019-11-02 at 9 12 37 AM

**Review by iccub — CHANGES_REQUESTED** [review]
Doesn't look well on ip8, I would leave it until 1.14, and for now just add some autoshrink to 9pt minimum or something

Zrzut ekranu 2019-11-2 o 16 14 23

**Comment by iccub** [comment]
Please address the feedback when you have time, we plan to ship it for 1.14

**Comment by Brandon-T** [comment]
> Please address the feedback when you have time, we plan to ship it for 1.14

Fixed. I had to change a bunch of other stuff in order for onboarding to look decent on iPhone 5S and iPhones without a notch as design was saying that the backgrounds are too low down on the screen (especially now that the buttons take up a larger vertical space now).

**Comment by Brandon-T** [comment]
iPhone 5S.. It now fits horizontally on all devices.

Screen Shot 2019-11-25 at 10 26 58 AM
Screen Shot 2019-11-25 at 10 26 52 AM
Screen Shot 2019-11-25 at 10 26 43 AM
Screen Shot 2019-11-25 at 10 26 37 AM

File: Client/Frontend/Browser/Onboarding/OnboardingAdsAvailableView.swift [hunk 1]
```diff
@@ -93,13 +95,13 @@ extension OnboardingAdsAvailableController {
...
 
-            [skipButton, continueButton, UIView.spacer(.horizontal, amount: 0)]
+            [skipButton, continueButton]
                 .forEach(buttonsStackView.addArrangedSubview(_:))
             
             [textStackView, buttonsStackView].forEach(descriptionStackView.addArrangedSubview(_:))
             
-            continueButton.snp.makeConstraints {
-                $0.centerX.equalTo(self.snp.centerX)
+            skipButton.snp.makeConstraints {
```
Comment (iccub) [inline Client/Frontend/Browser/Onboarding/OnboardingAdsAvailableView.swift:105]: Remove this and use `fillEqually` distribution for the button's stack view. Should work the same

File: Client/Frontend/Browser/Onboarding/OnboardingRewardsAgreementView.swift [hunk 2]
```diff
@@ -165,13 +167,13 @@ extension OnboardingRewardsAgreementViewController {
...
 
-            [skipButton, turnOnButton, UIView.spacer(.horizontal, amount: 0)]
+            [skipButton, turnOnButton]
                 .forEach(buttonsStackView.addArrangedSubview(_:))
             
             [textStackView, buttonsStackView].forEach(descriptionStackView.addArrangedSubview(_:))
             
-            turnOnButton.snp.makeConstraints {
-                $0.centerX.equalTo(self.snp.centerX)
+            skipButton.snp.makeConstraints {
```
Comment (iccub) [inline Client/Frontend/Browser/Onboarding/OnboardingRewardsAgreementView.swift:177]: see my other comment about `fillEqually`

File: Client/Frontend/Browser/Onboarding/OnboardingSearchEnginesView.swift [hunk 3]
```diff
@@ -111,19 +111,14 @@ extension OnboardingSearchEnginesViewController {
...
-            
             // Hiding views in prepration to animations.
             // Alpha is used instead of `isHidden` to make the views participate in auto-layout.
             [titlePrimary, titleSecondary, searchEnginesTable, buttonsStackView].forEach {
                 $0.alpha = CGFloat.leastNormalMagnitude
             }
             
-            continueButton.snp.makeConstraints {
-                $0.centerX.equalTo(self.snp.centerX)
+            skipButton.snp.makeConstraints {
```
Comment (iccub) [inline Client/Frontend/Browser/Onboarding/OnboardingSearchEnginesView.swift:122]: see my other comment about fillEqually

File: Client/Frontend/Browser/Onboarding/OnboardingSearchEnginesViewController.swift [hunk 4]
```diff
@@ -40,6 +40,17 @@ class OnboardingSearchEnginesViewController: OnboardingViewController {
         let tablebackground = UIView()
         tablebackground.backgroundColor = OnboardingViewController.colorForTheme(theme)
         contentView.searchEnginesTable.backgroundView = tablebackground
+        
+        // This selection should only ever happen once.
+        // We initially select the default search engine's cell
+        DispatchQueue.main.async {
+            for searchEngine in self.searchEngines.orderedEngines.enumerated() {
+                if searchEngine.element == self.searchEngines.defaultEngine() {
```
Comment (iccub) [inline Client/Frontend/Browser/Onboarding/OnboardingSearchEnginesViewController.swift]: You can remove this `if` and add it in foreach loop like this
`for searchEngine in self.searchEngines.orderedEngines.enumerated() where searchEngine.element == self.searchEngines.defaultEngine()` {
}

And split in two lines bc it's too long

File: Client/Frontend/Browser/Onboarding/OnboardingShieldsView.swift [hunk 5]
```diff
@@ -96,13 +98,13 @@ extension OnboardingShieldsViewController {
...
-            [skipButton, continueButton, UIView.spacer(.horizontal, amount: 0)]
-            .forEach(buttonsStackView.addArrangedSubview(_:))
+            [skipButton, continueButton]
+                .forEach(buttonsStackView.addArrangedSubview(_:))
             
             [textStackView, buttonsStackView].forEach(descriptionStackView.addArrangedSubview(_:))
             
-            continueButton.snp.makeConstraints {
-                $0.centerX.equalTo(self.snp.centerX)
+            skipButton.snp.makeConstraints {
```
Comment (iccub) [inline Client/Frontend/Browser/Onboarding/OnboardingShieldsView.swift:109]: see my other comment about it

**Review by iccub — CHANGES_REQUESTED** [review]

File: Client/Frontend/Browser/Onboarding/OnboardingAdsAvailableView.swift (hunk 1 above)
Comment (Brandon-T) [reply Client/Frontend/Browser/Onboarding/OnboardingAdsAvailableView.swift:105]: There's no `fillEqually` in my PR :S

File: Client/Frontend/Browser/Onboarding/OnboardingRewardsAgreementView.swift (hunk 2 above)
Comment (Brandon-T) [reply Client/Frontend/Browser/Onboarding/OnboardingRewardsAgreementView.swift:177]: Same.

File: Client/Frontend/Browser/Onboarding/OnboardingSearchEnginesView.swift (hunk 3 above)
Comment (Brandon-T) [reply Client/Frontend/Browser/Onboarding/OnboardingSearchEnginesView.swift:122]: Same

File: Client/Frontend/Browser/Onboarding/OnboardingShieldsView.swift (hunk 5 above)
Comment (Brandon-T) [reply Client/Frontend/Browser/Onboarding/OnboardingShieldsView.swift]: THere's no `fillEqually` in this PR.

File: Client/Frontend/Browser/Onboarding/OnboardingSearchEnginesViewController.swift (hunk 4 above)
Comment (Brandon-T) [reply Client/Frontend/Browser/Onboarding/OnboardingSearchEnginesViewController.swift]: Done.

File: Client/Frontend/Browser/Onboarding/OnboardingAdsAvailableView.swift (hunk 1 above)
Comment (Brandon-T) [reply Client/Frontend/Browser/Onboarding/OnboardingAdsAvailableView.swift:105]: Ahh I see what happened. The comment is older than the PR. I had already removed it before seeing this comment.

Comment (iccub) [reply Client/Frontend/Browser/Onboarding/OnboardingAdsAvailableView.swift:105]: It's better to use `fillEqually` distribution than to use constrains inside of stack view imo

**Review by iccub — APPROVED** [review]

**Pull request merged**